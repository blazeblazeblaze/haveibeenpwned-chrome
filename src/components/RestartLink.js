import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';

const RestartLink = ({onStartOver}) => {
  return (
    <Link onClick={onStartOver} className="pure-button pure-button-primary button-small">
      Try with a different email
    </Link>
  );
}

RestartLink.propTypes = {
  onStartOver: PropTypes.func.isRequired
};

export default RestartLink;
