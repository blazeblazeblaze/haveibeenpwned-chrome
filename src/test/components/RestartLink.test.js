import React from 'react';
import ReactDOM from 'react-dom';
import RestartLink from '../../components/RestartLink';
import renderer from 'react-test-renderer';


test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RestartLink onStartOver={jest.fn} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('RestartLink renders Link component', () => {
  const component = renderer.create(
   <RestartLink onStartOver={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
