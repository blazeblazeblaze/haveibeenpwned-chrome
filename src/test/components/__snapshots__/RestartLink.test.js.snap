// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestartLink renders Link component 1`] = `
<a
  className="pure-button pure-button-primary button-small"
  href=""
  onClick={[Function]}
>
  Try with a different email
</a>
`;
